services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -97,11 +97,16 @@ removeSettings (zusr ::: tid ::: _) = do
 getUserStatus :: UserId ::: TeamId ::: UserId ::: JSON -> Galley Response
 getUserStatus (_zusr ::: tid ::: uid ::: _) = do
     assertLegalHoldEnabled tid
-    teamMember <- Data.teamMember tid uid
-    case teamMember of
-        Nothing -> throwM teamMemberNotFound
-        Just member ->
-            pure . json $ UserLegalHoldStatusResponse (view legalHoldStatus member)
+    mTeamMember <- Data.teamMember tid uid
+    teamMember <- maybe (throwM teamMemberNotFound) pure mTeamMember
+    statusResponse <- case (view legalHoldStatus teamMember) of
+        UserLegalHoldDisabled ->
+            pure $ UserLegalHoldStatusResponse UserLegalHoldDisabled Nothing Nothing
+        status -> do
+            mLastKey <- fmap snd <$> LegalHoldData.selectPendingPrekeys uid
+            let mClientId :: Maybe ClientId = clientIdFromPrekey . unpackLastPrekey <$> mLastKey
+            pure $ UserLegalHoldStatusResponse status mLastKey mClientId
```

Comment:
I have little context here, but it looks iike you want to throw an exception here if one is `Just` and the other is `Nothing`.  Then you could change the data type to contain a tuple of both values inside a single `Maybe`.  Easier to read, and easier to reason about.  But I can't tell whether it's worth it.